hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataTableValidator.java

```diff
@@ -1730,13 +1743,28 @@ public List<HoodieColumnRangeMetadata<Comparable>> getSortedColumnStatsList(
       } else {
         FileFormatUtils formatUtils = HoodieIOFactory.getIOFactory(metaClient.getStorage())
             .getFileFormatUtils(HoodieFileFormat.PARQUET);
-        return fileNames.stream().flatMap(filename ->
-                formatUtils.readColumnStatsFromMetadata(
-                        metaClient.getStorage(),
-                        new StoragePath(FSUtils.constructAbsolutePath(metaClient.getBasePath(), partitionPath), filename),
-                        allColumnNameList).stream())
-            .sorted(new HoodieColumnRangeMetadataComparator())
-            .collect(Collectors.toList());
+        Schema readerSchema = new TableSchemaResolver(metaClient).getTableAvroSchema();
```

Comment:
can we reuse the schema available in the validation context?

Reply:
fixed this in my latest revision https://github.com/apache/hudi/pull/12159/commits/24dcbf67a0731d6213a2b8ae92241784e5da1c3f